### PR TITLE
Promote Operator check to public API; Strengthen composite Operator check

### PR DESCRIPTION
**Pull request by jeremylt**
Promote `CeedOperatorCheckReady` to public API

### CODE REVIEW

File: rust/libceed/src/operator.rs [hunk 1]
```diff
@@ -428,6 +428,12 @@ impl<'a> OperatorCore<'a> {
     }
 
     // Common implementations
+    pub fn check(&self) -> crate::Result<i32> {
+        let mut is_ready = false;
+        let ierr = unsafe { bind_ceed::CeedOperatorIsReady(self.ptr, &mut is_ready) };
+        self.check_error(ierr)
```
Comment (jedbrown) [inline rust/libceed/src/operator.rs:434]: Doesn't actually use the result `is_ready`

File: rust/libceed/src/operator.rs [hunk 2]
```diff
@@ -750,6 +756,50 @@ impl<'a> Operator<'a> {
...
+    /// // Check
+    /// assert!(op_build.check().is_ok(), "Operator not correctly setup");
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[allow(unused_mut)]
+    pub fn check(mut self) -> crate::Result<Self> {
+        self.op_core.check()?;
+        Ok(self)
+    }
```
Comment (jedbrown) [inline rust/libceed/src/operator.rs:801]: Why `mut`?

File: rust/libceed/src/operator.rs [hunk 3]
```diff
@@ -1993,6 +2043,62 @@ impl<'a> CompositeOperator<'a> {
...
+    ///     .sub_operator(&op_build_mass)?
+    ///     .sub_operator(&op_build_diff)?;
+    ///
+    /// // Check
+    /// assert!(op_build.check().is_ok(), "Operator not correctly setup");
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[allow(unused_mut)]
+    pub fn check(mut self) -> crate::Result<Self> {
```
Comment (jedbrown) [inline rust/libceed/src/operator.rs:2097]: And here; tests pass without it.

File: interface/ceed-operator.c [hunk 4]
```diff
@@ -907,6 +910,29 @@ int CeedCompositeOperatorAddSub(CeedOperator composite_op,
...
+  @ref Advanced
+**/
+int CeedOperatorIsReady(CeedOperator op, bool *is_ready) {
+  int ierr;
+
+  *is_ready = false;
+  ierr = CeedOperatorCheckReady(op); CeedChk(ierr);
+  *is_ready = true;
+
+  return CEED_ERROR_SUCCESS;
```
Comment (jedbrown) [inline interface/ceed-operator.c:933]: These are surprising semantics. One would expect to return with `is_ready=false` and no error code. `CeedOperatorCheckReady` provides more information than a bool (in the form of an error code). If we want that to be available in a non-error path, then perhaps the innermost function should be this non-error check. I'm skeptical that it's worth doing that -- a caller can just handle the error.

**Review by jedbrown — COMMENTED** [review]
May be useful, but semantics are weird now.

**Comment by jeremylt** [comment]
CI failure due to usual intermittent runner problem

**Review by jedbrown — APPROVED** [review]

**Pull request merged**